### PR TITLE
Geometry library

### PR DESCRIPTION
**Pull request by mgiulini**
You are about to submit a new Pull Request. Before continuing make sure you read the [contributing guidelines](CONTRIBUTING.md) and you comply with the following criteria:

- [x] You have stick to Python. Talk with us before adding other programming languages to HADDOCK3
- [ ] Your PR is about CNS
- [x] Your code is well documented: proper docstrings and explanatory comments for those tricky parts
- [x] You structured the code into small functions as much as possible. You can use classes if there's a (state) purpose
- [x] code follows our coding style
- [ ] You wrote tests for the new code
- [x] `tox` tests pass. *Run `tox` command inside the repository folder*
- [x] `-test.cfg` examples execute without errors. *Inside `examples/` run `python run_tests.py -b`*
- [x] PR does not add any *install dependencies* unless permission granted by the HADDOCK team
- [x] PR does not break licensing
- [x] Your PR is about writing documentation for already existing code :fire:
- [ ] Your PR is about writing tests for already existing code :godmode:

---

Created `libalign.py` to take care of sequence and structural alignments.

### CODE REVIEW

File: src/haddock/libs/libalign.py [hunk 1]
```diff
@@ -0,0 +1,505 @@
...
+
+def calc_rmsd(V, W):
+    """Calculate the RMSD from two vectors."""
+    diff = np.array(V) - np.array(W)
+    N = len(V)
+    return np.sqrt((diff * diff).sum() / N)
+
+
+def kabsch(P, Q):
+    """Find the rotation matrix using Kabsch algorithm."""
```
Comment (joaomcteixeira) [inline src/haddock/libs/libalign.py:84]: What is `P` and what is `Q`? Do they need to have any specific dimensions?

To render docs the best way, I would suggest:

```python
"""
Find the rotation matrix using Kabsch algorithm.

`Kabsch algorithm <https://en.wikipedia.org/wiki/Kabsch_algorithm>`_

Parameters
----------------
vec1 : np.array dtype=float, ndims=(3,), or compatible list
vec2 : np.array dtype=float, ndims=(3,), or compatible list

Returns
-----------
vec3 : np.array, dtype=float, ndims=?
    The resulting Kabsch calculation vector.
"""
```

I find it important to specify the input and output dimensions

**Review by joaomcteixeira — CHANGES_REQUESTED** [review]
Very nice! :rocket:
I gave a suggestion to enhance the docstrings. Which is actually the way to do it following numpydoc. If you render the docs `tox -e docs` you will see the suggested way is more detailed. 
You can see examples in the functions in this module I am preparing: https://github.com/haddocking/haddock3/blob/preprocessing/src/haddock/gear/preprocessing.py

Need to add the `.rst` files in the `docs/` folder. Can you? Following the examples? `docs/reference/libs/`.

Cheers!

File: src/haddock/libs/libalign.py (hunk 1 above)
Comment (mgiulini) [reply src/haddock/libs/libalign.py:84]: Agreed, I will try to setup a documentation for this library. Remember that this is quite preliminary, so the docs could change substantially in the near future

Comment (joaomcteixeira) [reply src/haddock/libs/libalign.py:84]: No worries if it is preliminary. Just write for what you have :wink:

Remember the docs should explain how to use the function without seeing the implementation code. For the style we use `numpydoc` style, yes.

You can try in your python prompt:

```python
from haddock.libs.libalign import kabsch


help(kabsch)
```

And see the outcome. Write the docstrings such that the help shows what you want to see :wink:

Great job!

**Review by rvhonorato — APPROVED** [review]

**Review by joaomcteixeira — APPROVED** [review]
I have corrected the lint. It was about giving some extra spaces and lines. Before I meant `ndims`, but I wanted to say `shape` - also corrected that.

Everything is good :+1: nice job!

**Comment by mgiulini** [comment]
cool, thanks! It's a bit crazy that we have to add a blank line in the middle of a function with no indentation to satisfy the lint tests, but I guess there's nothing we can do about it..

**Comment by joaomcteixeira** [comment]
It's not blank line. it is a spaceline between title headings and section contents. It's for letting the text breathe. You can always complain to `numpydoc` :-P I have invented nothing here, just following major conventions. I prefer using the empty line for spacing, otherwise is too crowded.

**Comment by mgiulini** [comment]
OK, I will adapt to it no problem ;)

**Pull request merged**